src/main/java/org/springframework/data/repository/query/Parameters.java

```diff
@@ -98,12 +98,12 @@ public Parameters(Method method) {
 	 */
 	protected Parameters(List<T> originals) {
 
-		this.parameters = new ArrayList<T>();
-
 		int pageableIndexTemp = -1;
 		int sortIndexTemp = -1;
 		int dynamicProjectionTemp = -1;
 
+		this.parameters = new ArrayList<T>(originals.size());
```

Comment:
In this case your right, this was just for "consistency" to the method above. Let me know when I should change it.